test/unit/puppetlabs/services/ca/certificate_authority_core_test.clj

```diff
@@ -351,15 +351,28 @@
           (is (= "Subject contains a wildcard, which is not allowed: foo*bar"
                  (:body response)))))
 
-      (testing "a CSR w/ DNS alt-names gets a specific error response"
-        (let [csr (io/input-stream (test-pem-file "hostwithaltnames.pem"))
-              response (handle-put-certificate-request!
-                        "hostwithaltnames" csr settings)]
-          (is (= 400 (:status response)))
-          (is (= (:body response)
-                 (str "CSR 'hostwithaltnames' contains subject alternative names "
-                      "(DNS:altname1, DNS:altname2, DNS:altname3), which are disallowed. "
-                      "Use `puppet cert --allow-dns-alt-names sign hostwithaltnames` to sign this request."))))))))
+     (testing "a CSR w/ DNS alt-names and disallowed subject-alt-names gets a specific error response"
```

Comment:
I don't think you need to add the `and disallowed subject-alt-names`, its the same thing as `DNS alt-names`, iiuc.

Reply:
So the difference here is that the csr has dns alt-names AND the setting says not to allow them.

Reply:
OOOooooooohhhhh

Reply:
Like allow-subject-alt-names settings if false